src/Rules.php

```diff
@@ -50,8 +50,8 @@ public function validate($value, ValidationContext $context = null): Result
             $ruleResult = $rule->validate($value, $context);
             if ($ruleResult->isValid() === false) {
                 $context->setParameter(self::PARAMETER_PREVIOUS_RULES_ERRORED, true);
-                foreach ($ruleResult->getErrors() as $message) {
-                    $compoundResult->addError($message);
+                foreach ($ruleResult->getErrors() as $key => $message) {
+                    $compoundResult->addError($message, $key);
```

Comment:
```suggestion
                    $compoundResult->addError($message, is_string($key) ? $key : null);
```

Seems, for number keys errors from diefferent rules can be overrided.

Reply:
Sorry. I didn't understand this one. How it can be overridden? Maybe provide an example? Also, in case of adding this, maybe move it inside `addError` method?

Reply:
Now I get it. :slightly_smiling_face:

Reply:
:heavy_check_mark: